### PR TITLE
Add UseMonoRuntime_LlvmJit option

### PR DESCRIPTION
**Pull request by fanyang-mono**
Add an option to support netcore Mono with LLVM JIT

### CODE REVIEW

File: src/Benchmarks.ServerJob/ServerJob.cs [hunk 1]
```diff
@@ -60,6 +60,7 @@ public class ServerJob : IIdentifiable
         public string RuntimeVersion { get; set; } = "";
         public string SdkVersion { get; set; } = "";
         public bool UseMonoRuntime { get; set; } = false;
+        public bool UseMonoRuntime_LlvmJit = false;
```
Comment (sebastienros) [inline src/Benchmarks.ServerJob/ServerJob.cs:63]: Maybe merge these two fields in one `string` containing the type of jit to use. I assume there might be a third value at some point. The option can still have no value to be compatible with the existing command line argument.

File: src/BenchmarksDriver/Program.cs [hunk 2]
```diff
@@ -747,6 +749,14 @@ public static int Main(string[] args)
                         Log("WARNING: '--self-contained' has been set implicitly as the mono runtime is used.");
                         Console.ResetColor();
                     }
+                    if (monoOption_LlvmJit.HasValue())
```
Comment (sebastienros) [inline src/BenchmarksDriver/Program.cs:752]: Just add an `||` with the previous statement. And if you merge both arguments then that works too.

File: src/Benchmarks.ServerJob/ServerJob.cs (hunk 1 above)
Comment (fanyang-mono) [reply src/Benchmarks.ServerJob/ServerJob.cs:63]: Yes, there will be a third flavor, which is AOT LLVM. What do you mean by merging those two fields but the option could still have no value? If we merge them, then it has to have a value to differentiate which flavor the use wants to have, right?

Comment (fanyang-mono) [reply src/Benchmarks.ServerJob/ServerJob.cs:63]: I got it now...

File: src/BenchmarksServer/Startup.cs [hunk 3]
```diff
@@ -3792,14 +3791,19 @@ private static void StartDotNetTrace(int processId, ServerJob job)
...
             {
-
-                var packageName = "Microsoft.NETCore.App.Runtime.Mono.linux-x64".ToLowerInvariant();
+                var packageName = "";
+                switch (mode) {
+                    case "jit":
+                        packageName = "Microsoft.NETCore.App.Runtime.Mono.linux-x64".ToLowerInvariant();
+                        break;
+                    case "llvm-jit":
+                        packageName = "Microsoft.NETCore.App.Runtime.Mono.LLVM.AOT.linux-x64".ToLowerInvariant();
```
Comment (SamMonoRT) [inline src/BenchmarksServer/Startup.cs:3804]: Is this right or should we use the LLVM.JIT package ? Seems confusing

Comment (fanyang-mono) [reply src/BenchmarksServer/Startup.cs:3804]: We don't have the LLVM.JIT package. For some reason, they never got generated. The literal only difference between the AOT LLVM and JIT LLVM packages are the AOT packages also contain llc/opt. So it is totally fine to use the AOT LLVM package for JIT LLVM.

**Comment by sebastienros** [comment]
LGTM. Have you tested the changes for the server?

**Comment by fanyang-mono** [comment]
> LGTM. Have you tested the changes for the server?

Yes, I've tested this change on Mono lab machines.

**Pull request merged**